### PR TITLE
Set 1 refactoring

### PR DESCRIPTION
**Pull request by Kabilesh020799**
I saw some code that could be refactored and performed it.
1.) Renamed the variable site to dataDogSite which gave more sense and was self-explaining, this could improve the code readability
2.) Introduced explaining variable for a complex if condition, that explains the purpose of the condition which in turn improves the code quality and code readability
3.) Found a code smell of complex method in the class Menufactory and splitter into 2 different methods so that the code readability and quality increase.

I kindly request you to merge these changes if you find this helpful. Thanks

### CODE REVIEW

File: jmeter-java-dsl-datadog/src/main/java/us/abstracta/jmeter/javadsl/datadog/DatadogBackendListener.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public static DatadogBackendListener datadogListener(String apiKey) {
    * @param site specifies the site to use. When no site is specified, US1 is used by default.
    * @return the listener for further configuration and usage.
    */
-  public DatadogBackendListener site(DatadogSite site) {
```
Comment (rabelenda) [inline jmeter-java-dsl-datadog/src/main/java/us/abstracta/jmeter/javadsl/datadog/DatadogBackendListener.java:61]: In general we are not adding the "set" prefix to methods for setting properties of test elements to keep them short and at this point follow the general convention. 

Additionally, I think that in this case adding Datadog to the method is redundant, since once you are setting a datadog property named site, I think is quite natural to think is the Datadog Site. I do agree though that datadog site is a little too generic, and doesn't easily reflect the purpose of the variable, maybe something like region would be better, but site is the name that datadog has assigned to it, so as not to confuse too much users we kept with that name. 

Finally, this is a non backwards change: if some users are already using the existing method, then their test plans will stop working unless they change the method. In general we avoid making these changes unless it was a very recent version that we hope none to too few users are using it. If we ever come to a change like this, and we really want to make it, then we put it in a backlog of changes to be done on next major version (eg 2.0).

File: jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/threadgroups/DslDefaultThreadGroup.java [hunk 2]
```diff
@@ -283,15 +283,19 @@ public DslDefaultThreadGroup holdIterating(String iterations) {
     addStage(new Stage(getLastStage().threadCount(), null, iterations));
     return this;
   }
-
   private void checkIterationsPreConditions() {
-    if (!(stages.size() == 1 && !ZERO.equals(stages.get(0).threadCount())
-        || stages.size() == 2 && ZERO.equals(stages.get(0).threadCount())
-        && !ZERO.equals(stages.get(1).threadCount()))) {
+    boolean isSingleStageHold = stages.size() == 1 && !ZERO.equals(stages.get(0).threadCount());
```
Comment (rabelenda) [inline jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/threadgroups/DslDefaultThreadGroup.java:287]: I like the spirit of this refactor! But what do you think of something like this?:

```java
boolean isJustRamp = stages.size() == 1 && !ZERO.equals(stages.get(0).threadCount());
boolean isJustDelayAndRamp = stages.size() == 2 && ZERO.equals(stages.get(0).threadCount())
        && !ZERO.equals(stages.get(1).threadCount()))
if (!(isJustRamp || isJustDelayAndRamp)) {
  throw new IllegalStateException("Holding for iterations is only supported after initial hold and ramp, or ramp.");
}
if (ZERO.equals(getLastStage().threadCount())) {
....
```

I think adding a variables for each component of the or is a good idea since the condition of the if is quite big otherwise. Then for the two remaining if conditions (negated or and comparing last stage thread count to zero) I think adding variables (and the additional code lines) is not worth it, since they are pretty simple and understandable.

File: jmeter-java-dsl/src/test/java/org/apache/jmeter/gui/util/MenuFactory.java [hunk 3]
```diff
@@ -44,6 +44,7 @@
 import org.apache.jmeter.visualizers.Printable;
 import org.apache.jorphan.gui.GuiUtils;
 import org.apache.jorphan.reflect.ClassFinder;
+import org.jetbrains.annotations.NotNull;
```
[the remark this answers is not in the document]
Comment (rabelenda) [reply jmeter-java-dsl/src/test/java/org/apache/jmeter/gui/util/MenuFactory.java:47]: We leave the IDE option open to any contributors as so anyone can use the IDE they are more comfortable with.

File: jmeter-java-dsl/src/test/java/org/apache/jmeter/gui/util/MenuFactory.java [hunk 4]
```diff
@@ -105,80 +106,95 @@ public final class MenuFactory {
 
   private static Set<String> classesToSkip() {
     return Arrays.stream(JMeterUtils.getPropDefault("not_in_menu", "").split(","))
-        .map(String::trim)
-        .collect(Collectors.toSet());
+            .map(String::trim)
```
Comment (rabelenda) [inline jmeter-java-dsl/src/test/java/org/apache/jmeter/gui/util/MenuFactory.java:109]: If you can please try to align the IDE code style to the project, or only submit changes that are not related with spacing to avoid back and forth of changes regarding this and also focus PR in the relevant changes.

File: jmeter-java-dsl/src/test/java/org/apache/jmeter/gui/util/MenuFactory.java [hunk 5]
```diff
@@ -105,80 +106,95 @@ public final class MenuFactory {
...
-          // JMeterTreeNode and TestBeanGUI are special GUI classes,
-          // and aren't intended to be added to menus
-          .filter(name -> !name.endsWith("JMeterTreeNode"))
-          .filter(name -> !name.endsWith("TestBeanGUI"))
-          .filter(name -> !name.equals("org.apache.jmeter.gui.menu.StaticJMeterGUIComponent"))
-          .filter(name -> !elementsToSkip.contains(name))
-          .distinct()
-          .map(String::trim)
-          .collect(Collectors.toList());
+      List<String> guiClasses = getModifiedStrings(elementsToSkip);
```
Comment (rabelenda) [inline jmeter-java-dsl/src/test/java/org/apache/jmeter/gui/util/MenuFactory.java:116]: As described in the class javadoc, this class is just a copy of an existing class from JMeter. In such classes that contain so many lines from original code, we avoid doing much refactor as to keep the class as similar to the original one and ease porting changes in the original code to the "modified" version of the class.

In any case, this class was added because we were experiencing some flakiness in some tests and modifying this class would allow us to better trace the issue. The tests have been stable for a long time now, so I think is time we just remove this class.

**Comment by rabelenda** [comment]
Thank you very much for reviewing the code and submitting this PR! I am really grateful of the effort that you have spent and I hope more people are encouraged to contribute by submitting PRs and any other way they can find to help us improve JMeter DSL.

I have left some comments on the PR, I hope you find them helpful and you keep sending PRs to help :).

Regards

**Comment by Kabilesh020799** [comment]
@rabelenda Thanks for the comments, they helped me to understand the code more and improved my knowledge on how to write test cases and do refactoring. Thank you so much for the guidance. I have raised a new pr with these changes
